src/base.ts

```diff
@@ -48,6 +48,7 @@ export class Base {
 
       return fetch(url, config).then((response) => {
         if (response.status < 300 && response.status >= 200) {
+          if (response.status === 204) return response.statusText as unknown as Promise<T>;
```

Comment:
Because a `204` request (should) have no body, do we need to make the `statusText` a type of `Promise<T>` or would it always be a `string` type? Do we ever expect `statusText` to not be a `string`?

If it's always going to be a `string` type, do we need to add the `as unknown as Promise<T>;`?